src/vm/methodtablebuilder.cpp

```diff
@@ -2826,198 +2851,191 @@ MethodTableBuilder::EnumerateClassMethods()
         // But first - minimal flags validity checks
         //
         // No methods in Enums!
-        if (fIsClassEnum)
-        {
-            BuildMethodTableThrowException(BFA_METHOD_IN_A_ENUM);
-        }
-        // RVA : 0
-        if (dwMethodRVA != 0)
+#ifndef _DEBUG // Don't run the minimal validity checks for the system dll (except in debug builds so we don't build a bad system dll)
```

Comment:
Should we use ` bmtProp->fNoSanityChecks` for this everywhere?

Reply:
That implies a subtle change in the ReadyToRun format around handling of generics, as its not guaranteed that the current logic that sets that flag from readytorun will verify this sorts of things. Of course, we may already have this problem.

Reply:
We assume that all types in R2R images with `READYTORUN_FLAG_SKIP_TYPE_VALIDATION` flag are well formed. I think it is fine to skip any validation of well-formed types.